consensus-types/blocks/getters.go

```diff
@@ -0,0 +1,589 @@
+}
+
+// PbBellatrixBlock returns the underlying protobuf object.
+func (b *SignedBeaconBlock) PbBellatrixBlock() (*eth.SignedBeaconBlockBellatrix, error) {
+	if b.version != version.Bellatrix {
+		return nil, errNotSupported("PbBellatrixBlock", b.version)
+	}
+	pb, err := b.Proto()
+	if err != nil {
+		return nil, err
+	}
+	return pb.(*eth.SignedBeaconBlockBellatrix), nil
+}
+
+// PbBlindedBellatrixBlock returns the underlying protobuf object.
+func (b *SignedBeaconBlock) PbBlindedBellatrixBlock() (*eth.SignedBlindedBeaconBlockBellatrix, error) {
+	if b.version != version.BellatrixBlind {
+		return nil, errNotSupported("PbBlindedBellatrixBlock", b.version)
+	}
+	pb, err := b.Proto()
+	if err != nil {
+		return nil, err
+	}
+	return pb.(*eth.SignedBlindedBeaconBlockBellatrix), nil
+}
+
+// Version of the underlying protobuf object.
+func (b *SignedBeaconBlock) Version() int {
+	return b.version
+}
+
+// Header converts the underlying protobuf object from blinded block to header format.
+func (b *SignedBeaconBlock) Header() (*eth.SignedBeaconBlockHeader, error) {
+	root, err := b.block.body.HashTreeRoot()
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not hash block body")
+	}
+
+	return &eth.SignedBeaconBlockHeader{
+		Header: &eth.BeaconBlockHeader{
+			Slot:          b.block.slot,
+			ProposerIndex: b.block.proposerIndex,
+			ParentRoot:    b.block.parentRoot,
+			StateRoot:     b.block.stateRoot,
+			BodyRoot:      root[:],
+		},
+		Signature: b.Signature(),
+	}, nil
+}
+
+// MarshalSSZ marshals the signed beacon block to its relevant ssz form.
+func (b *SignedBeaconBlock) MarshalSSZ() ([]byte, error) {
+	pb, err := b.Proto()
+	if err != nil {
+		return []byte{}, err
+	}
+	switch b.version {
+	case version.Phase0:
+		return pb.(*eth.SignedBeaconBlock).MarshalSSZ()
+	case version.Altair:
+		return pb.(*eth.SignedBeaconBlockAltair).MarshalSSZ()
+	case version.Bellatrix:
+		return pb.(*eth.SignedBeaconBlockBellatrix).MarshalSSZ()
+	case version.BellatrixBlind:
+		return pb.(*eth.SignedBlindedBeaconBlockBellatrix).MarshalSSZ()
+	default:
+		return []byte{}, errIncorrectBlockVersion
+	}
+}
+
+// MarshalSSZTo marshals the signed beacon block's ssz
+// form to the provided byte buffer.
+func (b *SignedBeaconBlock) MarshalSSZTo(dst []byte) ([]byte, error) {
+	pb, err := b.Proto()
+	if err != nil {
+		return []byte{}, err
+	}
+	switch b.version {
+	case version.Phase0:
+		return pb.(*eth.SignedBeaconBlock).MarshalSSZTo(dst)
+	case version.Altair:
+		return pb.(*eth.SignedBeaconBlockAltair).MarshalSSZTo(dst)
+	case version.Bellatrix:
+		return pb.(*eth.SignedBeaconBlockBellatrix).MarshalSSZTo(dst)
+	case version.BellatrixBlind:
+		return pb.(*eth.SignedBlindedBeaconBlockBellatrix).MarshalSSZTo(dst)
+	default:
+		return []byte{}, errIncorrectBlockVersion
+	}
+}
+
+// SizeSSZ returns the size of the serialized signed block
+func (b *SignedBeaconBlock) SizeSSZ() int {
+	pb, err := b.Proto()
+	if err != nil {
+		panic(err)
```

Comment:
All panics will be converted to errors once we update fastssz to generate `SizeSSZ (int, error)`